### PR TITLE
Include distributed tracing

### PR DESCRIPTION
**Pull request by rolfl**

### CODE REVIEW

**Comment by rolfl** [comment]
Include guidance on not insalling all instrumentation, only those you actually use

File: docs/operations/distributed-tracing.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Distributed tracing typically requires a mechanism for collecting traces (instrumentation)
+and a separate mechanism for reporting and visualizing those traces.
+
+* [OpenTelemetry](https://opentelemetry.io/) provides instrumentation in multiple languages
+including Node. OpenTelemetry is not yet released as a final stable version and is in Beta.
+OpenTelemetry replaces both OpenTracing and OpenCensus, neither of which are actively being
+developed (security fixes only).
+
+* [Jaeger](https://www.jaegertracing.io/) provides visualization of distributed traces.
```
Comment (mhdawson) [inline docs/operations/distributed-tracing.md:13]: I think we can move the discussion of the back ends to the guidance section as we've generally not made recommendations on that front, only the Node.js components.

File: docs/operations/distributed-tracing.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Distributed Tracing
+
+## Modules
```
Comment (mhdawson) [inline docs/operations/distributed-tracing.md:3]: We should update to use the same section headers, flow as the other sections. For example: https://github.com/nodeshift/nodejs-reference-architecture/blob/master/docs/functional-components/webframework.md

File: docs/operations/distributed-tracing.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+For non-trivial systems the benefits are typically worth the costs, especially when investigating
+problems.
+
+OpenTelemetry allows the ability to sample a subset of traces to reduce the storage and network
+utilization of the tracing data. The guidance here is that, where possible, sampling should be disabled
+such that all traces are recorded. If sampling is enabled, it is inevitable that the trace data is
+missing when trying to investigate some critical failure.
+
+## IBM Use
```
Comment (mhdawson) [inline docs/operations/distributed-tracing.md:55]: We've removed this section from the public facing docs in the nodeshift repo.

File: docs/operations/distributed-tracing.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+- additional services, and the related administration (Jaeger/ElasticSearch, etc.)
+
+For non-trivial systems the benefits are typically worth the costs, especially when investigating
+problems.
+
+OpenTelemetry allows the ability to sample a subset of traces to reduce the storage and network
+utilization of the tracing data. The guidance here is that, where possible, sampling should be disabled
+such that all traces are recorded. If sampling is enabled, it is inevitable that the trace data is
+missing when trying to investigate some critical failure.
+
```
Comment (mhdawson) [inline docs/operations/distributed-tracing.md:54]: I think there were also 2-3 specific things that came out of the discussion we had in terms of suggestions/recommendations on how to use OpenTelemetry/distributed tracing.

File: docs/operations/distributed-tracing.md (hunk 3 above)
Comment (rolfl) [reply docs/operations/distributed-tracing.md]: Change applied

File: docs/operations/distributed-tracing.md (hunk 2 above)
Comment (rolfl) [reply docs/operations/distributed-tracing.md]: Changes applied

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Discussed in last Ref Arch meeting, no concerns expressed. Landing

**Pull request merged**